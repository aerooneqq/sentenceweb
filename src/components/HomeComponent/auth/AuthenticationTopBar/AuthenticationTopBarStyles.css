#authenticationTopBarContainer{
    width: 100%;
    height: 60px;
    background: linear-gradient(56deg, #d3d3d3, #dfdfdf);

    border-radius: 5px;
    border-style: none;

    display: flex;
    flex-direction: row;
    align-items: center;

    text-align: center;
}

#authTopBarText{
    margin-left: 10px;
    font-family: Arial;
    font-size: 23px;
    color: white;
}

#authTopBarAppNameText {
    margin-left: 5px;
    font-family: Arial;
    font-size: 23px;
    color: orangered;
}

#authSlider{
  height: 48px;
  width: 30%;
  cursor: default;

  background: orangered;

  border-style: solid;
  border-width: 1px;
  border-color: orangered;
  border-radius: 5px;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  font-family: Arial;
  font-size: 20px;
  color: white;
}

#authSlider:hover {
  background-color: #FD591E;
}

.movingRightSlider{
  animation: movingRightSliderAnimation 0.7s ease forwards;
}

@keyframes movingRightSliderAnimation {
  to{
    margin-left: 65%;
  }
}

.movingLeftSlider{
  animation: movingLeftSliderAnimation 0.7s ease forwards;
}

@keyframes movingLeftSliderAnimation {
  from {
    margin-left: 65%;
  }

  to{
    margin-left: 5%;
  }
}
