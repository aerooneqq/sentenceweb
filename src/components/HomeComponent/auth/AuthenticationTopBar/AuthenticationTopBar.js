import React from "react"

//Styles
import "./AuthenticationTopBarStyles.css"

/**
 * This is the top bar of the registration/authentication component
 * 
 * PROPS LIST:
 * 1) mode - the mode of registration components (either signIn, signOut)
 */
export default class AuthenticationTopBar extends React.Component {
    constructor(props) {
        super(props);

        this.handleSliderClick = this.handleSliderClick.bind(this);
    }

    /**
     * Handles the slide click by invoking the fucntion which comes from
     * Authentication component which changes the "sign" mode (from "signIn" to "signUp" 
     * and vice versa)
     */
    handleSliderClick() {
      this.props.changeSignMode();
    }

    render() {
        return (
            <div id="authenticationTopBarContainer">
              <div id="authSlider" onClick={this.handleSliderClick}
                  className={this.props.mode === "signIn" ? "movingLeftSlider" :"movingRightSlider" }>
                      {this.props.mode === "signIn" ? "Sign in" : "Sign up"}
              </div>
            </div>
        )
    }
}
