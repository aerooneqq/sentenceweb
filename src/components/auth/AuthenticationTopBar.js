import React from "react"
import "./AuthenticationTopBarStyles.css"

export default class AuthenticationTopBar extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
          authSliderID: "authSlider"
        };

        this.handleSliderClick = this.handleSliderClick.bind(this);
    }

    handleSliderClick(){
      this.props.changeSignMode();
    }

    render() {
        return (
            <div id="authenticationTopBarContainer">
              <div id={this.state.authSliderID} onClick={this.handleSliderClick}
                  className={this.props.mode === "signIn" ? "movingLeftSlider" :"movingRightSlider" }>
                      {this.props.mode === "signIn" ? "Sign in" : "Sign up"}
              </div>
            </div>
        )
    }
}
