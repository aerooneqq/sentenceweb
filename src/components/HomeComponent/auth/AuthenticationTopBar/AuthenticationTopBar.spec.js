import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import AuthenticationTopBar from './AuthenticationTopBar';

describe("Authentication top bar", () => { 
     it("should render correctly", () => { 
        const props = { 
            mode: "signIn"
        }

        const authenticationTopBar = shallow(<AuthenticationTopBar {...props}/>);

        expect(shallowToJson(authenticationTopBar)).toMatchSnapshot();
     });

     it("should be in sign in mode when mode is sign in", () => { 
        const props = { 
            mode: "signIn"
        }

        const authenticationTopBar = shallow(<AuthenticationTopBar {...props} />)

        expect(authenticationTopBar.find("#authSlider").hasClass("movingLeftSlider")).toBe(true);
        expect(authenticationTopBar.find("#authSlider").text()).toBe("Sign in");
     });

     it("should be in sign up mode when the mode is sign up", () => { 
        const props = { 
            mode: "signUp"
        }

        const authenticationTopBar = shallow(<AuthenticationTopBar {...props} />);

        expect(authenticationTopBar.find("#authSlider").hasClass("movingRightSlider")).toBe(true);
        expect(authenticationTopBar.find("#authSlider").text()).toBe("Sign up");
     });
});