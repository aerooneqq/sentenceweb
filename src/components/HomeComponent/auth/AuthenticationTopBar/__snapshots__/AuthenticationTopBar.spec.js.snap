// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication top bar should render correctly 1`] = `
<div
  id="authenticationTopBarContainer"
>
  <div
    className="movingLeftSlider"
    id="authSlider"
    onClick={[Function]}
  >
    Sign in
  </div>
</div>
`;
